tests/Controller/ControllerTestTrait.php

```diff
@@ -75,6 +75,29 @@ public function authenticateAsAdherent(Client $client, string $emailAddress, str
         return $client->followRedirect();
     }
 
+    public function authenticateAsAdmin(Client $client, string $emailAddress, string $password): Crawler
```

Comment:
Just do 

```
public function authenticateAsAdmin(Client $client): void
    {
        $crawler = $client->request(Request::METHOD_GET, '/admin/login');

        $client->submit($crawler->selectButton('Connexion')->form([
            '_admin_email' => 'admin@en-marche-dev.fr',
            '_admin_password' => 'admin',
        ]));
    }
```

It's overkill to test the redirect and everything every time.

Reply:
@B-Galati some features are only visible when logged as a **super** admin. (e.g.: this very specific feature \o/)
`login` should be a parameter to handle every test case

Reply:
Alright, but you don't need to test redirects each time. You could use this method with a default value for the mail and then use the same password for admin and superadmin for convenience.